pkgs/coverage/bin/format_coverage.dart

```diff
@@ -130,6 +133,28 @@ Future<void> main(List<String> arguments) async {
     }
   }
   await outputSink.close();
+
+  // Check coverage against the fail-under threshold if specified
+  if (env.failUnder != null) {
```

Comment:
Something a little annoying about Dart is that the null safety inference doesn't quite work as expected for fields like this. Since it's technically possible that `failUnder` is a getter that sometimes returns null, the compiler can't infer that it's non-null inside this if statement. So to avoid all the `!` you have to store it in a local variable:

```Dart
final failUnder = env.failUnder;
if (failUnder != null) {
  // failUnder is known to be non-null in here, so we don't need ! anymore.
```